core/src/scene/importer.cpp

```diff
@@ -0,0 +1,257 @@
+        }
+
+        //set yaml node with normalized texture path
+        if (names.size() > 1) {
+            texture[index] = normTexPath;
+        } else {
+            texture = normTexPath;
+        }
+    }
+}
+
+void Importer::normalizeSceneTextures(Node& root, const std::string& parentPath) {
+
+    if (Node textures = root["textures"]) {
+        for (auto texture : textures) {
+            if (Node textureUrl = texture.second["url"]) {
+                setNormalizedTexture(textureUrl, {textureUrl.Scalar()}, parentPath);
+            }
+        }
+    }
+
+    if (Node styles = root["styles"]) {
+
+        for (auto styleNode : styles) {
+
+            auto style = styleNode.second;
+            //style->texture
+            if (Node texture = style["texture"]) {
+                if (texture.IsScalar()) {
+                    setNormalizedTexture(texture, {texture.Scalar()}, parentPath);
+                }
+            }
+
+            //style->material->texture
+            if (Node material = style["material"]) {
+                for (auto& prop : {"emission", "ambient", "diffuse", "specular", "normal"}) {
+                    if (auto propNode = material[prop]) {
+                        if (auto matTexture = material[propNode]["texture"]) {
+                            if (matTexture.IsScalar()) {
+                                setNormalizedTexture(matTexture, {matTexture.Scalar()}, parentPath);
+                            }
+                        }
+                    }
+                }
+            }
+
+            //style->shader->uniforms->texture
+            if (Node shaders = style["shaders"]) {
+                if (Node uniforms = shaders["uniforms"]) {
+                    for (auto uniformNode : uniforms) {
+                        StyleUniform styleUniform;
+                        auto uniformValue = uniformNode.second;
+                        if (SceneLoader::parseStyleUniforms(uniformValue, nullptr, styleUniform) &&
+                                styleUniform.type == "sampler2D") {
+                            if (styleUniform.value.is<UniformTextureArray>()) {
+                                setNormalizedTexture(uniformValue,
+                                             styleUniform.value.get<UniformTextureArray>().names,
+                                             parentPath);
+                            } else {
+                                auto name = styleUniform.value.get<std::string>();
+                                setNormalizedTexture(uniformValue, {name}, parentPath);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+bool Importer::loadScene(const std::string& path) {
+
+    auto scenePath = path;
+    auto sceneString = stringFromFile(scenePath.c_str(), PathType::resource);
+    auto sceneName = getFilename(scenePath);
+
+    if (m_scenes.find(sceneName) != m_scenes.end()) { return true; }
```

Comment:
Differentiating scenes by their file name alone seems inadequate. If a scene imports both `folder_a/scene.yaml` and `folder_b/scene.yaml` then by this logic, I think the second will be skipped. 

Also, if this is meant to be an early exit from the function, it should happen before the expensive call to `stringFromFile()`.